### PR TITLE
fix(settings): missing fields in api conversion and func renames

### PR DESCRIPTION
**Pull request by plyr4**
created_at was returning 0, only if the value was updated via API. 

also, noticed a variable type `newSettingps` so i made it the ol' shadow variable with an underscore

### CODE REVIEW

**Review by wass3rw3rk — APPROVED** [review]

File: api/types/settings/platform.go [hunk 1]
```diff
@@ -230,19 +230,23 @@ func (ps *Platform) SetUpdatedBy(v string) {
...
-	ps.SetCompiler(newSettingps.GetCompiler())
-	ps.SetQueue(newSettingps.GetQueue())
-	ps.SetRepoAllowlist(newSettingps.GetRepoAllowlist())
-	ps.SetScheduleAllowlist(newSettingps.GetScheduleAllowlist())
+	ps.SetCompiler(_ps.GetCompiler())
+	ps.SetQueue(_ps.GetQueue())
+	ps.SetRepoAllowlist(_ps.GetRepoAllowlist())
+	ps.SetScheduleAllowlist(_ps.GetScheduleAllowlist())
+
+	ps.SetCreatedAt(_ps.GetCreatedAt())
```
Comment (wass3rw3rk) [inline api/types/settings/platform.go:247]: `created_at` shouldn't be updateable, right?

**Review by wass3rw3rk — CHANGES_REQUESTED** [review]

File: api/types/settings/platform.go (hunk 1 above)
Comment (plyr4) [reply api/types/settings/platform.go:247]: no but that raises a good point this function should really be called `DuplicateInPlace` or something like that. i see the confusion

Comment (plyr4) [reply api/types/settings/platform.go:247]: or... `Overwrite` or `FromSettings` ?

fwiw its used in these places:
https://github.com/go-vela/server/blob/main/cmd/vela-server/server.go#L255
https://github.com/go-vela/server/blob/main/api/admin/settings.go#L121-L122

Comment (plyr4) [reply api/types/settings/platform.go:262]: i made some tweaks and renamed `Update` to `FromSettings`

**Comment by plyr4** [comment]
added a `settings.FromCLIContext` that fixes missing fields in `Restore`

**Review by ecrupper — APPROVED** [review]

**Review by wass3rw3rk — APPROVED** [review]

**Pull request merged**